clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -683,21 +686,65 @@ public Node provide() {
         }
     }
 
+    /**
+     * Provides context which the retry may refer to
+     */
+    class CallRetryContext {
+
+        private int tries = 0;
+        private long nextAllowedTryMs = 0;
+        private final double JITTER_MIN = 0.8;
+        private final double JITTER_MAX = 1.2;
+
+        public int tries() {
+            return tries;
+        }
+
+        public long nextAllowedTryMs() {
+            return nextAllowedTryMs;
+        }
+
+        private void updateTries(int currentTries) {
+            this.tries = currentTries + 1;
+        }
+
+        private void updateNextAllowTryMs() {
+            double jitter = Math.random() * (JITTER_MAX - JITTER_MIN) + JITTER_MIN;
+            int failures = tries - 1;
+            double exp = Math.pow(2, failures);
+            this.nextAllowedTryMs = time.milliseconds() +
+                    (long) Math.min(retryBackoffMaxMs, jitter * exp * retryBackoffMs);
+            // TODO: Remove the line below
+            System.out.println("nextAllow = " +  (long) Math.min(retryBackoffMaxMs, jitter * exp * retryBackoffMs));
+        }
```

Comment:
Similar logic exists in `ClusterConnectionStates.updateReconnectBackoff`. Maybe we could extract it to a utilities class.

Reply:
Make sense. Will extract it to a util class in my KIP-601 implementation.